source/views/home/edit/list.ios.js

```diff
@@ -61,9 +64,20 @@ class EditHomeView extends React.PureComponent<Props, State> {
     this.setState(() => ({width: event.window.width}))
   }
 
-  renderRow = ({data, active}: {data: ViewType, active: boolean}) => (
-    <EditHomeRow active={active} data={data} width={this.state.width} />
-  )
+  renderRow = ({data, active}: {data: ViewType, active: boolean}) => {
+    const enabled = !this.props.inactiveViews.includes(data.view)
+    return (
+      <EditHomeRow
+        active={active}
+        data={data}
+        isEnabled={enabled}
+        onToggle={()=>{
```

Comment:
We can "simplify" this: here, `onToggle={this.props.onToggleViewDisabled}`, and in Row.js, change `onToggleSwitch` to just call `this.props.onToggle(this.props.data.view)`, like you did before.